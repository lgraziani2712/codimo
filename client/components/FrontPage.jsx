/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';

import { COLOR_PALETTE } from 'core/constants/colors';

import ComputerThinkingURL from '../images/computer-thinking.png';

import PageContainer from './pages/PageContainer';
import Container from './pages/RawContentContainer';
import ImageGallery from './pages/ImageGallery';

const Row = styled.div`
  align-items: center;
  display: flex;
  min-height: 300px;
  padding: 5rem;
`;
const ColoredRow = styled(Row)`
  color: white;
  justify-content: flex-end;
  margin: 0;

  & > img {
    filter: drop-shadow(2px 2px 2px rgba(0, 0, 0, 0.5));
  }
`;
const RedRow = styled(ColoredRow)`
  background: ${COLOR_PALETTE.red.normal};
`;
const GreenRow = styled(ColoredRow)`
  background: ${COLOR_PALETTE.green.normal};
`;
const WhiteRow = styled(Row)`
  background: white;
  box-shadow: 0 0 4px 4px rgba(85, 95, 110, 0.4);
  justify-content: flex-start;
  position: relative;
  z-index: 1;

  & > img {
    filter: drop-shadow(-2px 2px 2px rgba(0, 0, 0, 0.5));
  }
`;
const Title = styled.h1`
  font-size: 4.5em;
  font-weight: 200;
  margin: 0;
  text-shadow: 5px 5px ${COLOR_PALETTE.red.dark};
`;
const Text = styled.div`
  font-family: 'Open Sans Condensed', sans-serif;
  font-size: 2em;
  font-weight: lighter;
  padding: 0 0.5em;
  width: 42%;
`;

const FrontPage = () => (
  <PageContainer>
    <RedRow>
      <Container>
        <Title>{'¡Donde el código, la imaginación y la motivación se encuentran!'}</Title>
      </Container>
    </RedRow>
    <WhiteRow>
      <img src="/images/logo.png" />
      <Text>
        {
          // eslint-disable-next-line max-len
          'Es una plataforma que presenta actividades escolares diseñadas para que sean divertidas y ayuden a afianzar los conocimientos de las distintas asignaturas a través de resolverlas de una manera diferente a como se aprenden en clase.'
        }
      </Text>
    </WhiteRow>
    <GreenRow>
      <Text>
        {
          // eslint-disable-next-line max-len
          'Cada actividad está estructurada en niveles. Cada nivel representa un desafío y a medida que se avanza de nivel, más aumenta la dificultad.'
        }
      </Text>
      <ImageGallery />
    </GreenRow>
    <WhiteRow>
      <img src={ComputerThinkingURL} />
      <Text>
        {'¿Qué es el '}
        <b>{'Pensamiento Computacional'}</b>
        {
          // eslint-disable-next-line max-len
          '? Es un conjunto de habilidades que le permite a una persona poder resolver problemas aplicando los conceptos fundamentales de la ciencia de la computación.'
        }
      </Text>
    </WhiteRow>
  </PageContainer>
);

export default FrontPage;
