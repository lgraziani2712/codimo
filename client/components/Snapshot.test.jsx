/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 * @jest-environment node
 */
import React from 'react';
import renderer from 'react-test-renderer';

import 'jest-styled-components';

import FrontPage from './FrontPage';

jest.mock('./pages/requireScreenshots');

describe('Components snapshots', () => {
  it('renders the FrontPage correctly', () => {
    const tree = renderer.create(<FrontPage />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
