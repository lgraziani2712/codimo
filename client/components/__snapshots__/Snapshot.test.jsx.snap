// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components snapshots renders the FrontPage correctly 1`] = `
.c11 {
  color: snow;
  font-weight: bold;
  text-decoration: none;
}

.c11:hover {
  color: #dadada;
  text-decoration: underline;
}

.c4 {
  margin: 0 auto;
  width: 980px;
}

.c0 {
  margin-bottom: 10rem;
  -webkit-user-select: initial;
  -moz-user-select: initial;
  -ms-user-select: initial;
  user-select: initial;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #428bca;
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  bottom: 0;
  height: 10rem;
  margin: 0;
  position: fixed;
  width: 100%;
  z-index: -1;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 300px;
  padding: 5rem;
}

.c2 {
  color: white;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin: 0;
}

.c2 > img {
  -webkit-filter: drop-shadow(2px 2px 2px rgba(0,0,0,0.5));
  filter: drop-shadow(2px 2px 2px rgba(0,0,0,0.5));
}

.c1 {
  background: #d9534f;
}

.c8 {
  background: #5cb85c;
}

.c6 {
  background: white;
  box-shadow: 0 0 4px 4px rgba(85,95,110,0.4);
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  position: relative;
  z-index: 1;
}

.c6 > img {
  -webkit-filter: drop-shadow(-2px 2px 2px rgba(0,0,0,0.5));
  filter: drop-shadow(-2px 2px 2px rgba(0,0,0,0.5));
}

.c5 {
  font-size: 4.5em;
  font-weight: 200;
  margin: 0;
  text-shadow: 5px 5px #be2f2b;
}

.c7 {
  font-family: 'Open Sans Condensed',sans-serif;
  font-size: 2em;
  font-weight: lighter;
  padding: 0 0.5em;
  width: 42%;
}

<div
  className="c0"
>
  .c3 {
  margin: 0 auto;
  width: 980px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 300px;
  padding: 5rem;
}

.c1 {
  color: white;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin: 0;
}

.c1 > img {
  -webkit-filter: drop-shadow(2px 2px 2px rgba(0,0,0,0.5));
  filter: drop-shadow(2px 2px 2px rgba(0,0,0,0.5));
}

.c0 {
  background: #d9534f;
}

.c7 {
  background: #5cb85c;
}

.c5 {
  background: white;
  box-shadow: 0 0 4px 4px rgba(85,95,110,0.4);
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  position: relative;
  z-index: 1;
}

.c5 > img {
  -webkit-filter: drop-shadow(-2px 2px 2px rgba(0,0,0,0.5));
  filter: drop-shadow(-2px 2px 2px rgba(0,0,0,0.5));
}

.c4 {
  font-size: 4.5em;
  font-weight: 200;
  margin: 0;
  text-shadow: 5px 5px #be2f2b;
}

.c6 {
  font-family: 'Open Sans Condensed',sans-serif;
  font-size: 2em;
  font-weight: lighter;
  padding: 0 0.5em;
  width: 42%;
}

<div>
    .c3 {
  margin: 0 auto;
  width: 980px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 300px;
  padding: 5rem;
}

.c1 {
  color: white;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin: 0;
}

.c1 > img {
  -webkit-filter: drop-shadow(2px 2px 2px rgba(0,0,0,0.5));
  filter: drop-shadow(2px 2px 2px rgba(0,0,0,0.5));
}

.c0 {
  background: #d9534f;
}

.c4 {
  font-size: 4.5em;
  font-weight: 200;
  margin: 0;
  text-shadow: 5px 5px #be2f2b;
}

<div
      className="c0 c1 c2"
    >
      .c0 {
  margin: 0 auto;
  width: 980px;
}

.c1 {
  font-size: 4.5em;
  font-weight: 200;
  margin: 0;
  text-shadow: 5px 5px #be2f2b;
}

<div
        className="c0"
      >
        .c0 {
  font-size: 4.5em;
  font-weight: 200;
  margin: 0;
  text-shadow: 5px 5px #be2f2b;
}

<h1
          className="c0"
        >
          ¡Donde el código, la imaginación y la motivación se encuentran!
        </h1>
      </div>
    </div>
    .c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 300px;
  padding: 5rem;
}

.c0 {
  background: white;
  box-shadow: 0 0 4px 4px rgba(85,95,110,0.4);
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  position: relative;
  z-index: 1;
}

.c0 > img {
  -webkit-filter: drop-shadow(-2px 2px 2px rgba(0,0,0,0.5));
  filter: drop-shadow(-2px 2px 2px rgba(0,0,0,0.5));
}

.c2 {
  font-family: 'Open Sans Condensed',sans-serif;
  font-size: 2em;
  font-weight: lighter;
  padding: 0 0.5em;
  width: 42%;
}

<div
      className="c0 c1"
    >
      <img
        src="/images/logo.png"
      />
      .c0 {
  font-family: 'Open Sans Condensed',sans-serif;
  font-size: 2em;
  font-weight: lighter;
  padding: 0 0.5em;
  width: 42%;
}

<div
        className="c0"
      >
        Es una plataforma que presenta actividades escolares diseñadas para que sean divertidas y ayuden a afianzar los conocimientos de las distintas asignaturas a través de resolverlas de una manera diferente a como se aprenden en clase.
      </div>
    </div>
    .c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 300px;
  padding: 5rem;
}

.c1 {
  color: white;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin: 0;
}

.c1 > img {
  -webkit-filter: drop-shadow(2px 2px 2px rgba(0,0,0,0.5));
  filter: drop-shadow(2px 2px 2px rgba(0,0,0,0.5));
}

.c0 {
  background: #5cb85c;
}

.c3 {
  font-family: 'Open Sans Condensed',sans-serif;
  font-size: 2em;
  font-weight: lighter;
  padding: 0 0.5em;
  width: 42%;
}

<div
      className="c0 c1 c2"
    >
      .c0 {
  font-family: 'Open Sans Condensed',sans-serif;
  font-size: 2em;
  font-weight: lighter;
  padding: 0 0.5em;
  width: 42%;
}

<div
        className="c0"
      >
        Cada actividad está estructurada en niveles. Cada nivel representa un desafío y a medida que se avanza de nivel, más aumenta la dificultad.
      </div>
      <img
        src="module.url"
      />
    </div>
    .c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 300px;
  padding: 5rem;
}

.c0 {
  background: white;
  box-shadow: 0 0 4px 4px rgba(85,95,110,0.4);
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  position: relative;
  z-index: 1;
}

.c0 > img {
  -webkit-filter: drop-shadow(-2px 2px 2px rgba(0,0,0,0.5));
  filter: drop-shadow(-2px 2px 2px rgba(0,0,0,0.5));
}

.c2 {
  font-family: 'Open Sans Condensed',sans-serif;
  font-size: 2em;
  font-weight: lighter;
  padding: 0 0.5em;
  width: 42%;
}

<div
      className="c0 c1"
    >
      <img
        src="computer-thinking.png"
      />
      .c0 {
  font-family: 'Open Sans Condensed',sans-serif;
  font-size: 2em;
  font-weight: lighter;
  padding: 0 0.5em;
  width: 42%;
}

<div
        className="c0"
      >
        ¿Qué es el 
        <b>
          Pensamiento Computacional
        </b>
        ? Es un conjunto de habilidades que le permite a una persona poder resolver problemas aplicando los conceptos fundamentales de la ciencia de la computación.
      </div>
    </div>
  </div>
  .c3 {
  color: snow;
  font-weight: bold;
  text-decoration: none;
}

.c3:hover {
  color: #dadada;
  text-decoration: underline;
}

.c2 {
  margin: 0 auto;
  width: 980px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #428bca;
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  bottom: 0;
  height: 10rem;
  margin: 0;
  position: fixed;
  width: 100%;
  z-index: -1;
}

<div
    className="c0"
  >
    .c2 {
  color: snow;
  font-weight: bold;
  text-decoration: none;
}

.c2:hover {
  color: #dadada;
  text-decoration: underline;
}

.c1 {
  margin: 0 auto;
  width: 980px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
}

<div
      className="c0 c1"
    >
      .c0 {
  color: snow;
  font-weight: bold;
  text-decoration: none;
}

.c0:hover {
  color: #dadada;
  text-decoration: underline;
}

<span>
        Equipo de desarrollo de Códimo. 
        .c0 {
  color: snow;
  font-weight: bold;
  text-decoration: none;
}

.c0:hover {
  color: #dadada;
  text-decoration: underline;
}

<a
          className="c0"
          href="https://www.unrn.edu.ar/"
          target="_blank"
        >
          UNRN
        </a>
        .
      </span>
      .c0 {
  color: snow;
  font-weight: bold;
  text-decoration: none;
}

.c0:hover {
  color: #dadada;
  text-decoration: underline;
}

<span>
        Distribuído bajo licencia 
        .c0 {
  color: snow;
  font-weight: bold;
  text-decoration: none;
}

.c0:hover {
  color: #dadada;
  text-decoration: underline;
}

<a
          className="c0"
          href="https://github.com/lgraziani2712/codimo/blob/master/LICENSE"
          target="_blank"
        >
          MIT
        </a>
        .
      </span>
    </div>
  </div>
</div>
`;
